recipe/meta.yaml

```diff
@@ -6,41 +6,55 @@ package:
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/quantile-forest-{{ version }}.tar.gz
-  sha256: 08bc4d733486292ae75c239dedc27723291e751d82fe8f372528ee043a608086
+  # Looks like the source code of v1.1.2 from PyPi is broken: file quantile_forest/version.txt cannot be found.
+  url: https://github.com/zillow/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: f2536c21797fde6ffffdb770e04d276d689de569ff40824361e52ad75b7fa97e
 
 build:
-  script: {{ PYTHON }} -m pip install . -vv
   number: 0
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
 
 requirements:
+  build:
+    - {{ compiler('cxx') }}
   host:
     - python
-    - cython >=3.0a4
-    - numpy >=1.23
-    - scipy >=1.4
-    - scikit-learn >=1.0
+    - cython 0.29.33
+    - numpy {{ numpy }}
     - pip
+    - scikit-learn
```

Comment:
why don't these host dependencies need exact pinning (i.e. removed >=1.0, >=1.4)?

Reply:
We have scikit-learn  with python 3.11 support:
```
scikit-learn                   1.1.3 py310h313beb8_1  pkgs/main
scikit-learn                   1.1.3 py310hc377ac9_0  pkgs/main
scikit-learn                   1.1.3 py311h313beb8_1  pkgs/main
scikit-learn                   1.1.3  py38h313beb8_1  pkgs/main
scikit-learn                   1.1.3  py38hc377ac9_0  pkgs/main
scikit-learn                   1.1.3  py39h313beb8_1  pkgs/main
scikit-learn                   1.1.3  py39hc377ac9_0  pkgs/main
scikit-learn                   1.2.0 py310h313beb8_0  pkgs/main
scikit-learn                   1.2.0 py310h313beb8_1  pkgs/main
scikit-learn                   1.2.0  py38h313beb8_0  pkgs/main
scikit-learn                   1.2.0  py38h313beb8_1  pkgs/main
scikit-learn                   1.2.0  py39h313beb8_0  pkgs/main
scikit-learn                   1.2.0  py39h313beb8_1  pkgs/main
scikit-learn                   1.2.1 py310h313beb8_0  pkgs/main
scikit-learn                   1.2.1 py311h313beb8_0  pkgs/main
scikit-learn                   1.2.1  py38h313beb8_0  pkgs/main
scikit-learn                   1.2.1  py39h313beb8_0  pkgs/main
scikit-learn                   1.2.2 py310h313beb8_0  pkgs/main
scikit-learn                   1.2.2 py310h313beb8_1  pkgs/main
scikit-learn                   1.2.2 py311h313beb8_0  pkgs/main
scikit-learn                   1.2.2 py311h313beb8_1  pkgs/main
scikit-learn                   1.2.2  py38h313beb8_0  pkgs/main
scikit-learn                   1.2.2  py38h313beb8_1  pkgs/main
scikit-learn                   1.2.2  py39h313beb8_0  pkgs/main
scikit-learn                   1.2.2  py39h313beb8_1  pkgs/main
```
But all of the versions have been rebuilt for different reasons. It's hard to decide which version to choose.

Reply:
I've pinned host dependencies:
```
    - scikit-learn 1.1.3
    - scipy 1.9.3
```

Reply:
But they failed. So it's not a good idea because they conflict with numpy